### PR TITLE
[Backup] Add Resume Protection functionality for Azure Virtual Machine

### PR DESCRIPTION
**Pull request by sambitratha**
**Description**  
Added Resume Protection functionality for Azure Virtual Machine

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S169

File: src/azure-cli/azure/cli/command_modules/backup/custom_base.py [hunk 1]
```diff
@@ -353,6 +353,8 @@ def resume_protection(cmd, client, resource_group_name, vault_name, container_na
     policy = show_policy(protection_policies_cf(cmd.cli_ctx), resource_group_name, vault_name, policy_name)
     custom_help.validate_policy(policy)
 
+    if item.properties.backup_management_type.lower() == "azureiaasvm":
+        return custom.resume_protection(cmd, client, resource_group_name, vault_name, item, policy)
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/backup/custom_base.py:357]: Use `elseif` for the condition branches below this one.

File: src/azure-cli/azure/cli/command_modules/backup/tests/latest/test_backup_commands.py [hunk 2]
```diff
@@ -55,7 +55,13 @@ def test_backup_scenario(self, resource_group, vault_name, vm_name, storage_acco
         self.kwargs['job'] = self.cmd('backup restore restore-disks -g {rg} -v {vault} -c {container} -i {item} -r {recovery_point} --storage-account {sa} --query name --restore-to-staging-storage-account').get_output_in_json()
         self.cmd('backup job wait -g {rg} -v {vault} -n {job}')
 
-        # Disable Protection
+        # Disable protection with retain data
+        self.cmd('backup protection disable -g {rg} -v {vault} -c {container} -i {item} --backup-management-type AzureIaasVM --workload-type VM --yes')
+
+        # Resume protection
+        self.cmd('backup protection resume -g {rg} -v {vault} -c {container} -i {item} --policy-name DefaultPolicy --backup-management-type AzureIaasVM')
+
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/backup/tests/latest/test_backup_commands.py:63]: No checks for the command results? How to verify the correctness of the command run?

File: src/azure-cli/azure/cli/command_modules/backup/custom_base.py (hunk 1 above)
Comment (sambitratha) [reply src/azure-cli/azure/cli/command_modules/backup/custom_base.py:357]: This is a pylint violation. It doesn't allow having elif statements when we have if blocks returning something. It throws this error when i run a style check.
`R1705: Unnecessary "elif" after "return" (no-else-return)`

Comment (fengzhou-msft) [reply src/azure-cli/azure/cli/command_modules/backup/custom_base.py:357]: Got it.

**Pull request merged**